src/websocket/FieldDetector.js

```diff
@@ -4,7 +4,9 @@ module.exports = class FieldDetector {
         this.configuredStreamIds = new Set()
     }
 
-    async detectAndSetFields(stream, streamMessage, apiKey, sessionToken) {
+    async detectAndSetFields(streamMessage, apiKey, sessionToken) {
+        const stream = await this.streamFetcher.fetch(streamMessage.getStreamId(), apiKey, sessionToken)
```

Comment:
as I remember this call was super slow, that's why we were caching this in `handlePublishRequest`.

Reply:
@mirpo `streamFetcher.fetch` already caches the results (with memoize), so to me it seemed that it was basically being double-cached for no obvious reason. Please let me know if I missed something!